### ISSUE TITLE
Macbook M1 `cargo build` issue inside `farms/farm-rpc` folder.

### ISSUE THREAD

**Issue by Prenden** [solana-labs/solana-program-library#3180]
When I try to build the farms/farm-rpc package with `cargo build` or `cargo build --release` I get the following error message.

```
error: failed to select a version for `solana-program`.
    ... required by package `anchor-lang v0.24.2`
    ... which satisfies dependency `anchor-lang = ">=0.17.0"` of package `quarry-mine v1.10.0`
    ... which satisfies dependency `quarry-mine = "^1.10.0"` of package `solana-farm-client v0.0.1 (/Users/prenden/Development/solana/solana-program-library/farms/farm-client)`
    ... which satisfies path dependency `solana-farm-client` of package `solana-farm-rpc v0.0.1 (/Users/prenden/Development/solana/solana-program-library/farms/farm-rpc)`
versions that meet the requirements `~1.9.13` are: 1.9.22, 1.9.21, 1.9.20, 1.9.19, 1.9.18, 1.9.17, 1.9.16, 1.9.15, 1.9.14, 1.9.13

all possible versions conflict with previously selected packages.

  previously selected package `solana-program v1.10.15`
    ... which satisfies dependency `solana-program = "=1.10.15"` of package `solana-address-lookup-table-program v1.10.15`
    ... which satisfies dependency `solana-address-lookup-table-program = "=1.10.15"` of package `solana-runtime v1.10.15`
    ... which satisfies dependency `solana-runtime = "=1.10.15"` of package `solana-transaction-status v1.10.15`
    ... which satisfies dependency `solana-transaction-status = "=1.10.15"` of package `solana-client v1.10.15`
    ... which satisfies dependency `solana-client = "^1.10.15"` of package `solana-farm-client v0.0.1 (/Users/prenden/Development/solana/solana-program-library/farms/farm-client)`
    ... which satisfies path dependency `solana-farm-client` of package `solana-farm-rpc v0.0.1 (/Users/prenden/Development/solana/solana-program-library/farms/farm-rpc)`

failed to select a version for `solana-program` which could resolve this conflict
```

<img width="1286" alt="Screenshot 2022-05-18 at 11 12 10" src="https://user-images.githubusercontent.com/22123409/169003356-6c0c43e5-13ca-4d49-9391-37db5c26a691.png">

**Comment by Prenden** [comment]
I have tried it with Ubuntu as well and that's failed too.

**Comment by Help-desk10** [comment]
Kindly open this live chat to talk to the customer service directly on your crypto issue  https://direct.lc.chat/14157678/

**Comment by askibin** [comment]
this will fix the issue when merged: https://github.com/solana-labs/solana-program-library/pull/3247

**Comment by joncinque** [comment]
Fixed with #3247
